wave_matching/include/wave/matching/matcher.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef __WAVE_MATCHING_MATCHER_HPP__
+#define __WAVE_MATCHING_MATCHER_HPP__
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+namespace wave {
+namespace matching {
+
+// sensor data type
+template <typename R>
+class Matcher {
+  public:
+    // generic accessors
+    const Eigen::Affine3d& get_result() {return result;};
+    const Eigen::MatrixXd& get_info() {return information;};
+    const float& getRes() {return resolution;};
```

Comment:
I would suggest to keep things simple.  Don't unecessarily pass by reference such a tiny thing.  Another optimization that is not warranted.